File: BackEnd/src/main/java/com/company/entity/cars/Car.java
package com.company.entity.cars;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "car")  // 테이블명이 car인 경우 명시
@Getter
@Setter
@NoArgsConstructor  // 기본 생성자 자동 생성
public class Car {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "car_model") // snake_case 사용 가능
    private String carModel;

    @Column(name = "image_url") // snake_case 사용 가능
    private String imageUrl;

    @Column(name = "car_make")
    private String carMake;

    // ✅ 기존 세터에서 오타 수정
    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }
}

File: GaragEZ/src/types/car.ts
// src/types/Car.ts
export interface Car {
  id: number; // ✅ Snake_Case → camelCase로 변경
  carModel: string;
  imageUrl: string;
  carMake: string;
  year: number; // ✅ 백엔드에 존재하는 필드 추가
  carNumber: string; // ✅ 추가된 필드
}
